apps/documentation/src/components/MdxProvider/MdxTitle/MdxTitle.tsx

```diff
@@ -1,15 +1,52 @@
-import type { ElementType, JSX, ReactNode } from 'react'
-import { Title, type TitleProps } from '@mantine/core'
+import {
+  useCallback,
+  type ElementType,
+  type JSX,
+  type MouseEvent,
+  type ReactNode,
+} from 'react'
+import { Title, Anchor, type TitleProps } from '@mantine/core'
+import { IconLink } from '@tabler/icons-react'
+import { useHover } from '@mantine/hooks'
 
 export default function MdxTitle(props: TitleProps): JSX.Element {
+  const { hovered, ref } = useHover<HTMLHeadingElement>()
+
+  const onLinkClick = useCallback(
+    (e: MouseEvent<HTMLAnchorElement>): void => {
+      e.preventDefault()
+      if (ref.current) {
+        ref.current.scrollIntoView({
+          behavior: 'instant',
+          block: 'start',
+        })
+      }
+    },
+    [ref],
```

Comment:
This handler should also update the location hash. 
At the moment it just perform a scroll into view of the title which isn't the behaviour I would expect from clicking an anchor link near a section title.

Reply:
I forgot to put it... good catch!